spinalcordtoolbox/deepseg/inference.py

```diff
@@ -107,6 +107,9 @@ def segment_monai_single(path_img, path_out, chkp_path, crop_size="64x192x-1", d
     Author: Naga Karthik
 
     """
+    # equivalent to `with torch.no_grad()`
+    torch.set_grad_enabled(False)
```

Comment:
My apologies for being unclear with my extended commit messages! I definitely should have provided much more context for why I made this change.

> My first guess would be, to remove one level of indentation

That's definitely one reason, but the other reason was memory usage (see below).

> And in this case, I guess we don't expect to sneakily affect other parts of the code with this setting (is this true?).

From my understanding, both `torch.no_grad` and `set_grad_enabled` [perform the same functionality](https://stackoverflow.com/a/53447634). So, to answer your question, I think we can focus strictly on the docs for [`torch.no_grad`](https://pytorch.org/docs/stable/generated/torch.no_grad.html):

> Disabling gradient calculation is useful for inference, when you are sure that you will not call [`Tensor.backward()`](https://pytorch.org/docs/stable/generated/torch.Tensor.backward.html#torch.Tensor.backward). It will reduce memory consumption for computations that would otherwise have `requires_grad=True`.
> 
> In this mode, the result of every computation will have `requires_grad=False`, even when the inputs have `requires_grad=True`. There is an exception! All factory functions, or functions that create a new Tensor and take a requires_grad kwarg, will NOT be affected by this mode.

My assumption was that there would be no computations involving back propagation /  `requires_grad=True` anywhere, given that we're loading (fixed) trained model weights? I also vaguely remember checking the outputs before/after the change and confirming that they were identical.

But, perhaps my change was too broad in scope, regardless? :thinking:

Reply:
> But, perhaps my change was too broad in scope, regardless? 🤔

I think, just to simplify review, I'm going to revert this change specifically (so that we don't have to second-guess any global changes).